### PR TITLE
linux: png app icons

### PR DESCRIPTION
**Pull request by Bloeckchengrafik**
This is a small pr that renders .png app icons on linux.

Svg from files doesn't seem to be supported by gpui currently. 
Do you have an idea on how to load these? If not, I'd wait with that until that's doable easily on gpui-side.

### CODE REVIEW

File: src/platform/linux/mod.rs [hunk 1]
```diff
@@ -23,12 +23,24 @@ pub fn get_app_data(path: &PathBuf) -> Option<AppData> {
         .to_string();
 
     let file = desktop_file::ApplicationDesktopFile::try_from(path).ok()?;
+    let icon_url: Option<PathBuf> = file.resolve_icon();
+
+    let icon_img = if let Some(icon) = icon_url.clone() {
+        if (icon.to_string_lossy().to_string().ends_with(".svg")) {
```
Comment (MatthiasGrandl) [inline src/platform/linux/mod.rs:29]: I would go with `icon.extension().unwrap_or_default().eq("svg")`

**Comment by Bloeckchengrafik** [comment]
This seems to use the default icons provided by the installed software and doesn't respect icon themes yet. 
I think this should be the topic of a future PR since there is no good way of getting theme information with gpui currently. Sure, we could add gtk or fltk as a dependency, but these are very heavy. Also, even the zed team hasn't decided on this yet. That's also one of the reasons why [this](https://github.com/zed-industries/zed/blob/0b34b1de7b74cf14d5aa03e8bb71272adad98e40/crates/gpui/src/platform/linux/platform.rs#L441) is currently not implemented if I heard it correctly.

**Comment by MatthiasGrandl** [comment]
@Bloeckchengrafik I'll start working on upstream GPUI SVG support as soon as I have the go-ahead from the Zed team. I think I figured out the proper way to do it.

**Comment by MatthiasGrandl** [comment]
@Bloeckchengrafik https://github.com/zed-industries/zed/pull/9436

**Pull request merged**